tcpclient/connection.go

```diff
@@ -1,14 +1,24 @@
 package tcpclient
 
-import "fmt"
+import (
+	"fmt"
+	"time"
+)
 
 type Connection struct {
-	ID     int
-	status connectionStatus
+	ID      int
+	status  connectionStatus
+	metrics connectionMetrics
 }
 
 type connectionStatus int
 
+type connectionMetrics struct {
+	timeTCPInitiatied  time.Time
```

Comment:
Does it make any sense to record when it was initiated and when it became establised? I don't see the objective, but just requiring how long it took to dial (so the difference between these times), but maybe you have other reports in mind that may benefit from this

Reply:
👍

Reply:
idea: what if we rename this to "processing time"? This is also valid for connections ending in errors (how long it took to generate the error?)

Reply:
ok, we can simply this with the duration of the TCP process, whatever the final state is.